k-distribution/src/main/scripts/lib/pyk/kastManip.py

```diff
@@ -529,24 +490,19 @@ def minimizeRule(rule, keepVars=[]):
         -   Unused cells will be abstracted.
         -   Attempt to remove useless side-conditions.
     """
-    if not (isKRule(rule) or isKClaim(rule)):
+    if not (type(rule) is KRule or type(rule) is KClaim):
         return rule
 
-    ruleBody = rule['body']
-    ruleRequires = rule['requires']
-    ruleEnsures = rule['ensures']
-    ruleAtts = rule['att']
+    ruleBody = rule.body
+    ruleRequires = rule.requires
+    ruleEnsures = rule.ensures
+    ruleAtts = rule.att
 
-    if ruleRequires is not None:
-        ruleRequires = buildAssoc(KToken('true', 'Bool'), '_andBool_', dedupe(flattenLabel('_andBool_', ruleRequires)))
-        ruleRequires = simplifyBool(ruleRequires)
-
-    if ruleEnsures is not None:
-        ruleEnsures = buildAssoc(KToken('true', 'Bool'), '_andBool_', dedupe(flattenLabel('_andBool_', ruleEnsures)))
-        ruleEnsures = simplifyBool(ruleEnsures)
+    ruleRequires = buildAssoc(KToken('true', 'Bool'), '_andBool_', dedupe(flattenLabel('_andBool_', ruleRequires)))
+    ruleRequires = simplifyBool(ruleRequires)
```

Comment:
Love it. So much more readable if we don't have to check for `None`s everywhere.